ecp5/arch.h

```diff
@@ -459,48 +487,47 @@ struct Arch : BaseCtx
     static bool is_available(ArchArgs::ArchArgsTypes chip);
     static std::vector<std::string> get_supported_packages(ArchArgs::ArchArgsTypes chip);
 
-    std::string getChipName() const;
+    std::string getChipName() const override;
     std::string get_full_chip_name() const;
 
-    IdString archId() const { return id("ecp5"); }
     ArchArgs archArgs() const { return args; }
```

Comment:
Nit: If these aren't override methods, should be `snake_case`?

Reply:
Actually looking through how these are used, they should become Arch API methods so I will add them as such

Reply:
I don't think I need to say this, but please update `archapi.md` if these are not already there.